frontend/src/app/messages/messages.component.ts

```diff
@@ -1,38 +1,47 @@
-import { HttpClient } from '@angular/common/http';
-import { Component, Inject } from '@angular/core';
-import { Observable } from 'rxjs/Observable';
+import { Component, Inject, OnInit, OnDestroy } from '@angular/core';
+import { ISubscription } from 'rxjs/Subscription';
 
-import { environment } from '../../environments/environment';
+import { MessagesService, Message } from './messages.service';
 import { WindowService } from '../window.service';
 
-interface Message {
-  message: string;
-  showMessage: boolean;
-  type: string;
-}
-
 @Component({
   selector: 'volontulo-messages',
-  templateUrl: './messages.component.html',
+  templateUrl: './messages.component.html'
 })
-export class MessagesComponent {
-  messages$: Observable<Message[]>;
-  showMessages = true;
-
-  constructor(
-    private http: HttpClient,
-        @Inject(WindowService) private window: any) {
-    this.messages$ = this.http.get<Message[]>(
-      `${environment.apiRoot}/messages/`).map(response => {
-      window.setTimeout(() => this.showMessages = false, 25000);
-      return response.map(message => {
-        message.showMessage = true;
-        return message;
+export class MessagesComponent implements OnInit, OnDestroy {
+  private messageSubscription: ISubscription;
+  public messages: Message[] = [];
+
+  constructor(private messagesService: MessagesService,
+    @Inject(WindowService) private window: any) { }
+
+  ngOnInit() {
+    this.messageSubscription = this.messagesService.message$
+      .subscribe((message: Message) => {
+        if (!message) {
```

Comment:
Isn't is here because subject starts from passing `null` to all its subscribers?

Reply:
Nope, it is there if you want to use clear() method and clear all the messages.